File: jediterm-core-mpp/src/jvmMain/java/com/jediterm/terminal/Questioner.java
package com.jediterm.terminal;

/**
 * @deprecated Collect extra information when creating {@link TtyConnector}.
 */
@SuppressWarnings("DeprecatedIsStillUsed")
@Deprecated(forRemoval = true)
public interface Questioner {
  String questionVisible(String question, String defValue);

  String questionHidden(String string);

  void showMessage(String message);
}

File: bossterm-core-mpp/src/jvmMain/kotlin/ai/rever/bossterm/terminal/Questioner.kt
package ai.rever.bossterm.terminal

/**
 * Interface for prompting user input during terminal connection setup.
 *
 * Used for interactive authentication flows like:
 * - SSH password authentication
 * - Two-factor authentication (2FA/MFA)
 * - Certificate passphrases
 */
interface Questioner {
    fun questionVisible(question: String?, defValue: String?): String?

    fun questionHidden(string: String?): String?

    fun showMessage(message: String?)
}
